log/namespaced/namespaces_test.go

```diff
@@ -0,0 +1,86 @@
+package namespaced
+
+import (
+	"testing"
+
+	. "github.com/smartystreets/assertions"
+)
+
+func TestNamespaces(t *testing.T) {
+	a := New(t)
+
+	ns := &ns{}
+
+	// empty namespace is accepted
+	{
+		ns.Set([]string{})
+
+		a.So(ns.IsEnabled("a"), ShouldBeFalse)
+		a.So(ns.IsEnabled("b"), ShouldBeFalse)
+		a.So(ns.IsEnabled("c"), ShouldBeFalse)
+		a.So(ns.IsEnabled(""), ShouldBeTrue)
+	}
+
+	// add some namespaces
+	{
+		ns.Set([]string{
+			"a",
+			"b",
+		})
+
+		a.So(ns.IsEnabled("a"), ShouldBeTrue)
+		a.So(ns.IsEnabled("b"), ShouldBeTrue)
+		a.So(ns.IsEnabled("c"), ShouldBeFalse)
+		a.So(ns.IsEnabled(""), ShouldBeTrue)
+	}
+
+	// * accepts everything
+	{
+		ns.Set([]string{
+			"*",
+		})
+
+		a.So(ns.IsEnabled("a"), ShouldBeTrue)
+		a.So(ns.IsEnabled("b"), ShouldBeTrue)
+		a.So(ns.IsEnabled("c"), ShouldBeTrue)
+		a.So(ns.IsEnabled(""), ShouldBeTrue)
+	}
+
+	// negation wins from *
+	{
+		ns.Set([]string{
+			"*",
+			"-a",
+		})
+
+		a.So(ns.IsEnabled("a"), ShouldBeFalse)
+		a.So(ns.IsEnabled("b"), ShouldBeTrue)
+		a.So(ns.IsEnabled("c"), ShouldBeTrue)
+		a.So(ns.IsEnabled(""), ShouldBeTrue)
+	}
+
+	// order should not matter
+	{
+		ns.Set([]string{
+			"-a",
+			"-b",
+			"*",
+		})
+
```

Comment:
I don't agree, the order should not matter when we specify that `-a` always wins (over `a` and `*`). This makes it simpler to reason about and it's also easier to implement.

In you case, what would `-a, -b, *` mean? Everything? But then why is the `-a` and `-b` there?